rotkehlchen/chain/evm/decoding/metamask/decoder.py

```diff
@@ -0,0 +1,177 @@
+            return DEFAULT_DECODING_OUTPUT
+
+        out_event = in_event = None
+        for event in context.decoded_events:
+            asset_symbol = event.asset.resolve_to_asset_with_symbol().symbol
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND or
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == sender
+            ):
+                # find the send event
+                event.counterparty = CPT_METAMASK
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.notes = f'Swap {event.balance.amount} {asset_symbol} in {CPT_METAMASK} from {event.location_label}'  # noqa: E501
+                event.address = self.router_address
+                out_event = event
+            elif event.event_type == HistoryEventType.RECEIVE and event.location_label == sender:
+                # find the receive event
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.counterparty = CPT_METAMASK
+                event.notes = f'Receive {event.balance.amount} {asset_symbol} from {CPT_METAMASK} swap in {event.location_label}'  # noqa: E501
+                event.address = self.router_address
+                # use this index as the event may be a native currency transfer
+                # and appear at the start
+                event.sequence_index = context.tx_log.log_index
+                in_event = event
+                to_amount = event.balance.amount
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND and
+                event.counterparty in AMM_POSSIBLE_COUNTERPARTIES
+            ):
+                # It is possible that in the same transaction we find events
+                # decoded by another amm such as uniswap and then the metamask decoder
+                # (as it appears at the end). In those cases we need to
+                # take the out event as the other amm event
+                out_event = event
+
+        # maybe reshuffle out first then in
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
+
+        # extract the fee info
+        fee_raw = fee_asset = None
+        for log in context.all_logs:
+            if log.topics[0] == METAMASK_FEE_TOPIC:
+                fee_raw = hex_or_bytes_to_int(log.data)
+                fee_asset = self.native_currency
+                break
+            if (
+                log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                log.topics[2] == self.gnosis_safe_proxy
```

Comment:
AS mentioned before this is an address, so decode it as one and compare to it

```suggestion
                hex_or_bytes_to_address(log.topics[2]) == self.gnosis_safe_proxy
```